### Objectif 

This notebook explain how to scraping data from website, for this task, I will use BeautifulSoup from bs4 packages, as supplementary , and Selenium package to do the same think.

# USING BeautifulSoup

### Load necessary packages

In [1]:
import requests 
import re
from bs4 import BeautifulSoup

### Extracting html code for one page

In [3]:
# The link, extract  html code just for one page 
link = 'https://www.indeed.fr/cmp/IBM/reviews?fcountry=ALL'
r = requests.get(link) # Because Beautiful Soup is not an HTTP client...
soup = BeautifulSoup(r.text,'html.parser')
#print(soup.prettify())

In [4]:
# Extract all the data for the first 100 pages 
x1=1
x2=1170
nbr_pages = [x*20 for x in range(x1, x2)]

links = []
rs    = []
soup  = []

for i in nbr_pages:
    links.append(link+'&start='+str(i)) 
for j in range(0,len(links)):
    rs.append(requests.get(links[j]))
    soup.append(BeautifulSoup(rs[j].text,'html.parser'))

In [4]:
# The second page containt this html code
# print(soup[0].prettify())

In [5]:
soup[0].find_all('div',class_='cmp-review-pro-text')[0].get_text()

'Work from home'

In [6]:
#cmp-review-title          #  div 
#cmp-reviewer-job-title    # span
#cmp-reviewer-job-location # span
#cmp-review-date-created   # span
#cmp-review-description    # div 

title = []
job_title = []
location = []
date = []
desc = []


for j in range((len(soup)-1)):
    for i in range(len(soup[j].find_all('div',class_='cmp-review-title'))):
        title.append(soup[j].find_all('div',class_='cmp-review-title')[i].get_text())
        job_title.append(soup[j].find_all('span',class_='cmp-reviewer-job-title')[i].get_text())        
        location.append(soup[j].find_all('span',class_='cmp-reviewer-job-location')[i].get_text())        
        date.append(soup[j].find_all('span',class_='cmp-review-date-created')[i].get_text())  
        desc.append(soup[j].find_all('div',class_='cmp-review-description')[i].get_text())

In [ ]:
# Create dataframe with four columns

In [9]:
import pandas as pd 
df = pd.DataFrame([date,title,job_title,location,desc]).T
df.columns = ['date','title','job_title','location','desc']
df.head()

date                                           title  \
0   9 octobre 2018                       travailleur salle blanche   
1  5 novembre 2018                              Good place to work   
2  5 novembre 2018                                 trabajo externo   
3  5 novembre 2018                     Conocimiento y proactividad   
4  5 novembre 2018  Fabulous place to work and develop your skills   

                                           job_title  \
0         Ouvrier de production (ancien employé) –     
1        Account Specialist/SDM (ancien employé) –     
2  Analista TI proyecto office 365 (ancien employ...   
3              Business Analyst (ancien employé) –     
4  National Sales Executive, IBM Software Group (...   

                                  location  \
0                   Coudray-Montceaux (91)   
1                     Bengaluru, Karnataka   
2          San Nicolás de los Garza, N. L.   
3  Santiago de Chile, Región Metropolitana   
4                              Ontario, CA   

                                                desc  
0  bonne expérience, travail en salle blanche par...  
1  It is a good place to work. Work/Life balance ...  
2  buen ambiente de trabajo aprendi administrar l...  
3  Jornada flexible, aprendi a crear soluciones a...  
4  People are very professional but the company's...

In [10]:
df.shape

(378, 5)

wouuw , as you can see, we have a data frame with 378 rows or reviews and 5 columns,the deal is to understand the reviews, I have to analyse the textuel data and see if the reviews of salaries how worked in the past in IBM are satisfaid or not. we remark that there is a review in diffrente language, in the next step (notebook), I will show you how to extract just english reviews and how to use NLP to describe and study the reviews of IBM company 